src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSPathogenReferenceTaxonProperties.java

```diff
@@ -11,10 +14,34 @@
     public String name = null;
     public String rank = null;
     public String parentTaxId = null;
-    public long length = 0;
-    public final Set<String> accessions;
+    private long length = 0;
```

Comment:
Not a comment on the `length` and `accessions` fields, but on the name and rank fields: is it possible to make them immutable  (final). I looked at where they are written, and it seems like they are written only when the default ctor of this class is used and the `name` and `rank` are immediately put after that. What about having a ctor that takes this two arguments?